earth_enterprise/src/fusion/autoingest/AssetGenPublicCpp.pl

```diff
@@ -85,6 +86,26 @@ ()
     void GetConfig(DOMElement *elem, $config &config);
 }
 
+
+
+namespace{
```

Comment:
Just thinking out loud: it would sure be nice if we could combine some of this duplicated code for assets and asset versions. That's true here and in several other places. Perhaps we could have a common base class for both? If we did that we could even go to a single Storage Manager, which would make managing memory usage easier.

To be clear, I don't want you do make this change on this PR. It would be too much. It's an interesting idea, though.

Reply:
I'm with you there. I'd love to have a common base for them.